python/cudf/cudf/pylibcudf_tests/test_string_find.py

```diff
@@ -0,0 +1,118 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.
+
+import numpy as np
+import pyarrow as pa
+import pytest
+from utils import assert_column_eq
+
+import cudf._lib.pylibcudf as plc
+from cudf._lib.scalar import DeviceScalar
+
+
+@pytest.fixture(scope="module")
+def string_col():
+    return pa.array(
+        ["AbC", "de", "FGHI", "j", "kLm", "nOPq", None, "RsT", None, "uVw"]
+    )
+
+
+@pytest.mark.parametrize("target", ["a", ""])
+def test_find(string_col, target):
+    plc_col = plc.interop.from_arrow(string_col)
+    got = plc.strings.find.find(
+        plc_col, DeviceScalar(target, dtype=np.dtype("object")).c_value, 0, -1
+    )
+    expected = pa.Array.from_pandas(
+        string_col.to_pandas().str.find(target), type=pa.int32()
+    )
+    assert_column_eq(got, expected)
+
+
+def test_find_column(string_col):
+    plc_col = plc.interop.from_arrow(string_col)
+    target_col = plc.interop.from_arrow(
+        pa.array(["A", "d", "F", "j", "k", "n", None, "R", None, "u"])
+    )
+    expected = pa.array([0, 0, 0, 0, 0, 0, None, 0, None, 0], type=pa.int32())
+    got = plc.strings.find.find(plc_col, target_col, 0)
+    assert_column_eq(got, expected)
+
+
+@pytest.mark.parametrize("target", ["a", ""])
+def test_rfind(string_col, target):
+    plc_col = plc.interop.from_arrow(string_col)
+    got = plc.strings.find.rfind(
+        plc_col, DeviceScalar(target, dtype=np.dtype("object")).c_value, 0, -1
+    )
+    expected = pa.Array.from_pandas(
+        string_col.to_pandas().str.rfind(target), type=pa.int32()
+    )
+    assert_column_eq(got, expected)
+
+
+@pytest.mark.parametrize("target", ["a", "aB", "Ab", ""])
+def test_contains(string_col, target):
+    plc_col = plc.interop.from_arrow(string_col)
+    got = plc.strings.find.contains(
+        plc_col, DeviceScalar(target, dtype=np.dtype("object")).c_value
+    )
+    expected = pa.Array.from_pandas(
+        string_col.to_pandas().str.contains(target)
+    )
```

Comment:
You should be able to check [`count_substring(...) > 1`](https://arrow.apache.org/docs/python/generated/pyarrow.compute.count_substring.html#pyarrow.compute.count_substring) roughly.